mp2-common/src/eth.rs

```diff
@@ -111,6 +116,175 @@ pub struct ProofQuery {
     pub(crate) slot: StorageSlot,
 }
 
+/// Struct used for storing relevant data to query blocks as they come in.
+#[derive(Debug, Clone)]
+pub struct ReceiptQuery {
+    /// The contract that emits the event we care about
+    pub contract: Address,
+    /// The event we wish to monitor for,
+    pub event: Event,
+}
+
+/// Struct used to store all the information needed for proving a leaf in the Receipt Trie is one we care about.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ReceiptProofInfo {
```

Comment:
All the `Info` struct, i'm a bit wondering whether they fit here or not.
To me the querying in this module should return what alloys return or maybe something a bit more digested, but the circuits should be handling that conversion to all the information that it needs. For example, only the circuit needs to know the relative offset of where the address starts. This is not a "general info" about a specific receipt. So it feels it suits better in the receipt circuit API imo.

@nicholas-mainardi wdyt ?

Reply:
Okay, I'll move it into `mp2-v1/src/values_extraction/receipt_leaf.rs`